### PR TITLE
Phpstan issues

### PR DESCRIPTION
**Pull request by mathieuimbert**
## Changes

- Upgrade phpstan version
- Fix issues reported by phpstan

## Issue

https://github.com/deployphp/deployer/issues/2996

- [x] Bug fix #…?
- [ ] New feature?
- [ ] BC breaks?
- [ ] Tests added?
- [ ] Docs added?

      Please, regenerate docs by running next command:
      $ php bin/docgen

### CODE REVIEW

**Comment by antonmedv** [comment]
Both unit and e2e tests are failing.

**Comment by mathieuimbert** [comment]
> Both unit and e2e tests are failing.

True :thinking: But I see they were already failing in the latest master: https://github.com/deployphp/deployer/runs/5413879503?check_suite_focus=true. It seems to be separate issues, don't you think?

**Comment by antonmedv** [comment]
Ohh, true. Forgot to merge https://github.com/deployphp/deployer/pull/2995

**Comment by antonmedv** [comment]
Can you rebase on master, please?

File: src/functions.php [hunk 1]
```diff
@@ -201,12 +201,10 @@ function task(string $name, $body = null): Task
             // There is no "up" or "down"casting in PHP.
             throw new \Exception('Tried to replace a Task with a GroupTask or vice-versa. This is not supported. If you are sure you want to do that, remove the old task `Deployer::get()->tasks->remove(<taskname>)` and then re-add the task.');
         }
-        if ($existingTask instanceof Task) {
-            $existingTask->setCallback($body);
-        } elseif ($existingTask instanceof GroupTask) {
+        if ($existingTask instanceof GroupTask) {
             $existingTask->setGroup($body);
-        } else {
-            throw new \LogicException('Unexpected Task type.');
```
Comment (antonmedv) [inline src/functions.php:209]: This looks like a real found bug. Awesome!

**Review by antonmedv — APPROVED** [review]

**Pull request merged**